### ISSUE TITLE
Wasm exhibiting unexptected output when data and policy package clash

### ISSUE THREAD

**Issue by nickhiggs** [open-policy-agent/opa#3386]
## Expected Behavior

The OPA wasm behaviour performs the same as the standard rego behaviour.

## Actual Behavior

Wasm fails to return the expected output when there is data & policy package that clashes. See the below example: 

## Steps to Reproduce the Problem

Setup the following files:
```
--- foo.rego
package foo

enabled = true

--- bar.rego
package bar

default out = "fail"
out = data.foo[input.a]

--- data.json
{
  "foo": {
    "b": true
  }
}

--- input.json
{"a": "b"}

```

and run:
```
go run -tags opa_wasm main.go eval -d foo.rego -d bar.rego -d data.json -i input.json data.bar.out -t wasm
```
is expected to return `true` but returns `fail` where as without running with `-t wasm` it returns `true`

using the master branch commit `ee9dc917` (latest master as of today).

**Comment by srenatus** [comment]
This looks like it might get fixed by #3344. I'll verify it later today if you don't beat me to it.

**Comment by srenatus** [comment]
✔️ I've verified it, #3344 will indeed do the trick here, too.
